Tests/BSONTests/DocumentTests.swift

```diff
@@ -0,0 +1,65 @@
+import BSON
+import Foundation
+import Nimble
+
+final class DocumentTests: BSONTestCase {
+    // This is a test in itself, will fail to compile on unsupported values
+    static let testDoc: BSONDocument = [
+        "int": 0xBAD1DEA,
+        "int32": .int32(32),
+        "int64": .int64(64)
+    ]
+
+    func testInt32Encoding() {
```

Comment:
do this and the next test give us any coverage we aren't getting now that the int32/int64 corpus tests are running?